chain/chaintest/balance_handler.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/codec/codectest"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/state/tstate"
+)
+
+// TestBalanceHandler tests b by requiring that it upholds the invariants
+// described in the BalanceHandler interface.
+func TestBalanceHandler(ctx context.Context, t *testing.T, b chain.BalanceHandler) {
+	// Test addresses
+	addrOne := codectest.NewRandomAddress()
+	addrTwo := codectest.NewRandomAddress()
+
+	tests := []struct {
+		Name string
+
+		Actor  codec.Address
+		Amount uint64
+
+		// State keys are not enforced in Setup
+		Setup func(context.Context, *testing.T, state.Mutable)
+		// State keys are enforced in TestFunc
+		TestFunc func(context.Context, codec.Address, state.Mutable, uint64) error
+		// Assertion is run after TestFunc for any additional checks
+		// State keys are not enforced in Assertion
+		Assertion func(context.Context, *testing.T, state.Mutable)
+
+		ExpectedBalance uint64
+		ExpectErr       bool
+	}{
+		{
+			Name:   "simpleAddBalance",
```

Comment:
We can avoid camel-casing here - these strings are just a UX improvement when running tests so we should make these human readable. We can also give more descriptive names like "add balance - overflow" for example.

Reply:
Done